File: exercism/clojure/grains/src/grains.clj
(ns grains)

(defn exp [x n]
  (loop [acc 1 n n]
    (if (zero? n) acc
        (recur (* x acc) (dec n)))))


(defn square [n] ;; <- arglist goes here
    ;; your code goes here
  (biginteger (exp 2N (dec n))) 
)

(defn total []  ;; <- arglist goes here
    ;; your code goes here
  (- (exp 2N 64) 1)
)

File: exercism/fortran/grains/grains.f90

module grains

   implicit none

contains

   double precision function square(n)
      integer :: n
      if(n >= 1 .and. n <= 64) then
         square = 2d0 ** (n - 1)
      else
         square = -1d0
      end if
   end function

   double precision function total()
      total = (2.d0 ** 64) - 1
   end function

end module
